tree/treeplayer/src/TTreeFormula.cxx

```diff
@@ -796,7 +796,7 @@ Int_t TTreeFormula::ParseWithLeaf(TLeaf* leaf, const char* subExpression, bool f
       }
 
       TBranch *bmom = branch->GetMother();
-      if( bmom->IsA() == TBranchElement::Class() )
+      if( bmom->IsA() == TBranchElement::Class() && br)
```

Comment:
Thanks for the PR again, thanks for coming back to this so late!

@ferdymercury, you know I'm all in for fixing warnings etc., but this time I don't agree with the suggestion.

Let's look at this `&& br` example. Clearly, there is some logic error in the code if `br == nullptr`, because inside the `if` branch, `br` is de-referenced and you'd get a crash. So at least you learn about the logic error, albeit in not the nicest way.

But now, the `if` will just be skipped, and who knows what consequences this will have? Maybe you even get silently the wrong results, just for the sake of avoiding a crash!

In other words, I don't think it's ok to just skip code if there is an accidental `nullptr` and continue running when there was a clear logic error.

I think there are four ways to proceed here:

1) Study the code to make sure that `br == nullptr` is indeed *not* a logic error, but a corner case for which skipping the `if` is the right behavior.
2) Throw an exception if `br == nullptr` as a nicer alternative to the crash you'd get with current ROOT `master`.
3) See if the warning can be fixed by changing the code at another place, that doesn't imply a change of behavior
4) Leave things as they are (close this PR), since we also haven't received related bug reports. Not all clang-tidy warnings are relevant, some are false positives.

I think
1) Might be a lot of work for unclear benefit
2) Maybe, but I feat the code will be overly verbose and inefficient if we add `if(x == nullptr) throw std::runtime_error("x was unexpectedly nullptr")` everywhere
3) Also quite some work, maybe not relevant given that we're not sure that these paths that clang-tidy outlined will ever happen
4) If I list multiple options, of course I have a preferred one, and in this case it's the "do nothing" option 😂